pkg/headers/headers.go

```diff
@@ -0,0 +1,118 @@
+package headers
+
+import (
+	"bufio"
+	"fmt"
+	"golang.org/x/text/cases"
+	"golang.org/x/text/language"
+	"io/ioutil"
+	"os"
+	"path/filepath"
+	"strings"
+)
+
+func Headers_entrypoint() {
+	scan(".")
+}
+func readfile(file string) (string, error) {
+
+	content, err := ioutil.ReadFile(file)
+	if err != nil {
+		return "", err
+	}
+	return string(content), nil
+
+}
+func generateAttachmentHeaders(content string) (string, error) {
+	scanner := bufio.NewScanner(strings.NewReader(content))
+
+	result := ""
+	for scanner.Scan() {
+		if strings.HasPrefix(scanner.Text(), "![") {
+			attachmentPath := strings.Split(strings.Split(scanner.Text(), "(")[1], ")")[0]
+
+			result += fmt.Sprintf("<!-- Attachment: %s -->\n", attachmentPath)
+		} else {
+			continue
+		}
+	}
+	return result, nil
+
+}
+func generateTitle(path string) string {
+	paths := strings.Split(path, "/")
+	title := strings.Replace(paths[len(paths)-1], ".md", "", -1)
+
+	return fmt.Sprintf("<!-- Title: %s -->\n", cases.Title(language.English).String(title))
+}
+func generateParentsHeaders(path string) ([]string, error) {
+	parents := strings.Split(path, "/")
+	parents = parents[0 : len(parents)-1]
+	for i := 0; i < len(parents); i++ {
+		parents[i] = fmt.Sprintf("<!-- Parent: ADR-%s -->\n", parents[i])
+	}
+	return parents, nil
+}
+func modifyHeader(path string) (string, error) {
+	result := ""
+	content, file_err := readfile(path)
+	if file_err != nil {
+		return "", file_err
+	}
+
+	scanner := bufio.NewScanner(strings.NewReader(content))
+
+	default_header, header_err := os.ReadFile("./scripts/default-header")
```

Comment:
If somebody deletes these files, the synchronization stops working.
I suggest defining constants that will hold default values.
If there is a chance the default values change often, I'd add a couple of CLI arguments to let a user decide where to get these values from.